homeassistant/components/metoffice/data.py

```diff
@@ -65,14 +76,36 @@ async def async_update(self):
     def _update(self):
         """Get the latest data from DataPoint."""
         if self._site is None:
-            _LOGGER.error("No Met Office forecast site held, check logs for problems")
+            _LOGGER.error("No Met Office sites held, check logs for problems")
             return
 
         try:
-            forecast = self._datapoint.get_forecast_for_site(
+            forecast_3hourly = self._datapoint.get_forecast_for_site(
```

Comment:
According to MartinHjelmare's comments below about disabling entities, it might be best to have two different coordinators to fetch data according to which entity is enabled.  This would reduce the amount of IO done if one of the entities is disabled.  I don't see this as critical however.